src/System.Text.Primitives/System/Text/Encoders/EncodingHelper.cs

```diff
@@ -56,18 +56,9 @@ public static bool IsWhitespace(uint codePoint)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static bool IsSupportedCodePoint(uint codePoint)
+        public static bool IsValidScalarValue(uint codePoint)
         {
-            if (codePoint >= FirstNotSupportedCodePoint)
-                return false;
-            if (codePoint >= HighSurrogateStart && codePoint <= LowSurrogateEnd)
-                return false;
-            if (codePoint >= 0xFDD0 && codePoint <= 0xFDEF)
-                return false;
-            if (codePoint == 0xFFFE || codePoint == 0xFFFF)
-                return false;
-
-            return true;
+            return (codePoint < FirstNotSupportedCodePoint) && !IsSurrogate(codePoint);
```

Comment:
This is an incorrect change for the way that this function is being used in our encoders. It doesn't consider the small extra holes we check in the code you removed.

Reply:
I'd have to dig back through emails (changes) on why we did this, but I vaguely remember there being a reason.